app/models/post.rb

```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :user
+
+  def update_posts_counter
+    user.increment!(:posts_counter)
+  end
```

Comment:
[Optional]

- I highly suggest that you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` to call the `#update_posts_counter` method every time a new post has been saved to the database, then make the `update_posts_counter` method private. Similarly, you can do this with `update_comments_counter` and `update_likes_counter` as well.